woocommerce/storefront
# Add basic Travis CI config for running e2e tests on PRs

Part of #1330

This adds a basic travis config and necessary package.json dependencies and scripts for running e2e tests.

### How to test the changes in this Pull Request:
Test the scripts locally:

- `npm run e2e:start` should bring up a wp-env environment, on http://localhost:8801 (dev) and http://localhost:8802 (tests).
- `npm run e2e:stop` should take that environment down.
- `npm run e2e:destroy` should take the environment down and destroy container & database state.
- `npm run e2e` should run the e2e tests; requires that the test environment is running.
- `npm run e2e:ci` should bring up the test environment and run end-to-end tests against it.

Review the Travis config. The goal of this config is to run e2e tests against open PRs as a blocking check.

As far as I can tell, Travis will only start building once the config is merged to `master` branch. Tutorial says: [`commit and push to trigger a Travis CI build`](https://docs.travis-ci.com/user/tutorial/). I'm going to merge this PR to test this - will revert if there are problems.

### Changelog

> Dev – Added basic end-to-end tests. #1408
